src/Swarm/Game/Step.hs

```diff
@@ -2138,11 +2144,19 @@ execConst c vs s k = do
     [WorldUpdate Entity] ->
     [RobotUpdate] ->
     m CESK
-  finishCookingRecipe r v wf rf = do
-    time <- use ticks
-    let remTime = r ^. recipeTime
-    return . (if remTime <= 1 then id else Waiting (remTime + time)) $
-      Out v s (FImmediate c wf rf : k)
+  finishCookingRecipe r v wf rf =
+    if remTime <= 0
+      then do
+        mapM_ (\(ReplaceEntity loc _ newEntity) -> updateEntityAt loc $ const newEntity) wf
```

Comment:
Why not call the `updateWorld` function here?  This lambda seems dangerous since it will become partial if we ever add other kinds of world updates.